### PR TITLE
Inherit the dependencies from the workspace

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

It's possible to define the dependency version & features in `workspace Cargo.toml` instead of in every `crates Cargo.toml`.
Now the `crate's toml`  file can inherit the dependency version and feature from the `workspace Cargo.toml` .
For example
```
In `workspace Cargo.toml`

[workspace.dependencies]
serde = "1.0"

In the crate's Cargo.toml

`serde = { workspace =true }`.
```
This will make it easy to update the version.
The `<crate>.workspace = true` syntax is not supported by cargo sort, So, used <crate> = { workspace =true }

This PR only addresses the dependencies that were defined multiple times.

**Todo**

The following instances use different versions of crates for some features, to use the same version, the code needs to be changed.

- Clap version that is used in tedge
- Mockito in tedge-http-ext
- toml in tedge-config

- Define the dependencies that are defined only once

- We can also define the local crate dependencies in the Workspace Cargo.toml and inherit from the crate's cargo.toml.
For example
```
in Workspace Cargo.toml
tedge_actors = { path = "crates/core/tedge_actors" }

and inherit in the project's Cargo.toml as below
tedge_actors = { workspace = true }
```

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2181

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/common/download/Cargo.toml [hunk 1]
```diff
@@ -12,21 +12,21 @@ repository = { workspace = true }
 [dependencies]
 anyhow = { version = "1.0", features = ["backtrace"] }
```
Comment (didier-wenzek) [inline crates/common/download/Cargo.toml:13]:
```suggestion
anyhow = { workspace = true, features = ["backtrace"] }
```

File: crates/common/log_manager/Cargo.toml [hunk 2]
```diff
@@ -11,13 +11,13 @@ repository = { workspace = true }
 [dependencies]
 easy_reader = "0.5"
 glob = "0.3"
```
Comment (didier-wenzek) [inline crates/common/log_manager/Cargo.toml:13]: I would also consider to move the version to the workspace even for those dependencies used only once.

**Review by didier-wenzek — COMMENTED** [review]
Thank you. This will ease our dependency management.

It would be good to also include in this PR:
- the dependencies that are defined only once
- local crate dependencies

It makes sense to address in specific PRs the following exceptions as the code must be upgraded:

- clap version that is used in tedge
- mockito in tedge-http-ext
- toml in tedge-config

**Comment by PradeepKiruvale** [comment]
> Thank you. This will ease our dependency management.
> 
> It would be good to also include in this PR:
> 
> * the dependencies that are defined only once
> * local crate dependencies
> 
> It makes sense to address in specific PRs the following exceptions as the code must be upgraded:
> 
> * clap version that is used in tedge
> * mockito in tedge-http-ext
> * toml in tedge-config

I agree with the plan. So, I have updated the PR addressing the first two points. i.e the dependencies that are defined once and other local crate dependencies.

File: crates/common/log_manager/Cargo.toml (hunk 2 above)
Comment (PradeepKiruvale) [reply crates/common/log_manager/Cargo.toml]: fixed

File: crates/common/download/Cargo.toml (hunk 1 above)
Comment (PradeepKiruvale) [reply crates/common/download/Cargo.toml]: fixed

File: Cargo.toml [hunk 3]
```diff
@@ -11,7 +11,7 @@ members = [
     "plugins/c8y_remote_access_plugin",
     "plugins/tedge_apt_plugin",
     "plugins/tedge_dummy_plugin",
-    "plugins/tedge_log_plugin",
+    "plugins/tedge_log_plugin",
```
Comment (didier-wenzek) [inline Cargo.toml:14]:
```suggestion
    "plugins/tedge_log_plugin",
```

File: crates/extensions/tedge_log_manager/Cargo.toml [hunk 4]
```diff
@@ -10,29 +10,29 @@ homepage = { workspace = true }
...
-anyhow = "1.0"
-assert-json-diff = "2.0"
-filetime = "0.2"
+anyhow = { workspace = true }
+assert-json-diff = { workspace = true }
+filetime = { workspace = true }
 tedge_actors = { path = "../../core/tedge_actors", features = ["test-helpers"] }
 tedge_http_ext = { path = "../../extensions/tedge_http_ext", features = [
     "test_helpers",
 ] }
```
Comment (didier-wenzek) [inline crates/extensions/tedge_log_manager/Cargo.toml:36]: These two have to be updated too.

**Review by didier-wenzek — APPROVED** [review]
Approved. Thank you.

File: crates/extensions/tedge_log_manager/Cargo.toml (hunk 4 above)
Comment (PradeepKiruvale) [reply crates/extensions/tedge_log_manager/Cargo.toml]: done

File: Cargo.toml (hunk 3 above)
Comment (PradeepKiruvale) [reply Cargo.toml]: fixed

**Pull request merged**